### PR TITLE
Parallelize queue successful message deletion

### LINKED ISSUE #4147: Parallelize the queue message deletion inside QueueBatchedReportStreamUploader
### Description
Currently we're blocking on each message deletion from the queue which causes any deletion slowdown to be increased linearly by the number of the deleted messages

### Proposed solution 
Parallelize the message deletion by not blocking on the deleted message

### Additional context
https://github.com/CDCgov/prime-simplereport/blob/a54ff56fefe977314201ac180556fc691ae04e77/ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts#L138

https://github.com/CDCgov/prime-simplereport/blob/a54ff56fefe977314201ac180556fc691ae04e77/ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts#L152

### PR DESCRIPTION
**Pull request by zdeveloper**
# Azure Function PULL REQUEST

## Related Issue

- fixes #4147

## Changes Proposed

- Parallelize the deletion of the successful message from the queue
- Update dependencies to fix failing tests

## Additional Information

- hardcoded `node-fetch` to  `2.6.6` since the lastest 2.6.7 (as of right now) is causing the following import error
`node_fetch_1.Headers is not a constructor`
- considered upgrading to `node-fetch v3`, but thats a wsi only package and it seems like it might need a of work to migrate since its not compatible with `CommonJS`. perhaps we should move to axios if node-fetch v2 stops being supported.
- added `package-lock.json` to lock the dependencies
- tested on dev

## Testing

- How should reviewers verify this PR?
- deploy branch to any dev env
- login to azure, submit some tests, verify records in the queue, observe the function app logs and trace when the records get removed

## Checklist for Primary Reviewer

- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke tested
- [X] Any content updates (user-facing error messages, etc) have been approved by content team
- [X] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [X] Any dependencies introduced have been vetted and discussed
- [X] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

File: ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts [hunk 1]
```diff
@@ -141,36 +142,48 @@ export async function deleteSuccessfullyParsedMessages(
...
+      message.popReceipt
+    ));
+  }
 
-      if(message.dequeueCount > 1){
+  try {
+    const promiseValues = await Promise.allSettled(deletionPromises);
+    for (let i = 0; i < promiseValues.length; i++) {
+      const promise = promiseValues[i];
+      const message = messages[i];
```
Comment (mehansen) [inline ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts:172]: if deletion is skipped, the messages and promiseValues arrays won't necessarily line up, right?

**Review by mehansen — COMMENTED** [review]
anything in particular I should look for when reviewing these dependency updates besides testing that it doesn't break my local or our test envs?

File: ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts (hunk 1 above)
Comment (zdeveloper) [reply ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts:172]: great catch! that is correct, will fix it asap

**Comment by zdeveloper** [comment]
> anything in particular I should look for when reviewing these dependency updates besides testing that it doesn't break my local or our test envs?

The only real dependency update was Jest, all the others were minor number adjustments and due to the caret in the version, these were the real values that we were using, assuming the package-lock was blown away.

Another weird situation was the `package-lock.json` was a v2 format which means it was generated with an npm that was higher than v6, which ultimately means node version higher than 14. I believe we're using node 14 in prod, [azure runtime 3.x which uses node 14](https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=azure-cli%2Cin-process%2Cv4&pivots=programming-language-typescript#:~:text=GA%20(Node.js%2014%2C%2012%2C%20%26%2010)).

**Review by mehansen — APPROVED** [review]

**Review by emmastephenson — APPROVED** [review]

**Review by DanielSass — APPROVED** [review]

**Pull request merged**